src/extension.ts

```diff
@@ -2,16 +2,19 @@ import * as vscode from "vscode";
 import graphFactories from "./util/graphFactories";
 import { Graph } from "./typings/Types";
 import makeGraph, { FactoryMap } from "./util/makeGraph";
-import { logBranchTypes } from "./util/debug";
+import { enableDebugLog, logBranchTypes } from "./util/debug";
 import { ThatMark } from "./core/ThatMark";
 import { TestCaseRecorder } from "./testUtil/TestCaseRecorder";
 import { getCommandServerApi, getParseTreeApi } from "./util/getExtensionApi";
 import isTesting from "./testUtil/isTesting";
 import CommandRunner from "./core/commandRunner/CommandRunner";
+import { ExtensionMode } from "vscode";
 
 export async function activate(context: vscode.ExtensionContext) {
   const { getNodeAtLocation } = await getParseTreeApi();
   const commandServerApi = await getCommandServerApi();
+  const isModeDevelop = context.extensionMode === ExtensionMode.Development;
```

Comment:
I like to be able to have this output active even not in develop mode.  I'd advocate for a setting and command to toggle debug output on or off

Reply:
Is it enough with one or the other or do we need both?

Reply:
I would lean towards both, because I will personally have them on all the time, and in other situations I'd just ask users to turn it on to check something.  It's what we have for decorations, where there's a setting that determines whether they're shown by default, and then a command to quickly toggle them on / off.

But I don't feel strongly.  Just having one or the other is prob good enough if you think it's too much noise to have both

Reply:
I think that if you're going to be running it all the time and we want something simple for the users to turn on in debugging purposes a setting is probably the more appropriate. 
I assume this setting will only be active in production mode? It shouldn't enable debugging in test for example.

Reply:
Added a setting called debug